### PR TITLE
Support for multiple oidc providers

### PR DESCRIPTION
**Pull request by tehsis**
## Overview

We want to use a different OIDC provider for teleport (for instance auth0.com) which currently works as is with Teleport.
This PR adds support for a `Display` field on the OIDC provider configuration to use in the login button on the web ui. So you can use a friendly name to identify the provider (which may be google, auth0 or even your specific company name).

The PR also allows you to use different providers on the same auth cluster (you can use google and auth0 or two or more different google/auth0 accounts)

## Example configuration

```yaml
auth_service:
  enabled: yes
  cluster_name: cluster-b
  trusted_clusters:
      - key_file: /var/lib/teleport/cluster-a
  oidc_connectors:    
      - id: auth0
        display: Auth0
        redirect_url: https://10.0.10.20:3080/web/login:3080/v1/webapi/oidc/callback
        client_id: <auth0-client-id>
        client_secret: <auth0-client-secret>
        issuer_url: https://tehsis-teleport.auth0.com/ 
      - id: auth0-eu
        redirect_url: https://10.0.10.20:3080/web/login:3080/v1/webapi/oidc/callback
        client_id: <auth0-client-id>
        client_secret: <auth0-client-secret>
        issuer_url: https://tehsis-eu.eu.auth0.com/
```

This result in the following ui:

screen shot 2016-11-25 at 5 34 34 pm

Note that since display is not specified on the second connection, the id is used as a default value.

### CODE REVIEW

**Review by klizhentas — APPROVED** [review]
Looks good to me, @alex-kovoy can you review UI part as well?

**Comment by klizhentas** [comment]
ok to test

**Comment by klizhentas** [comment]
ok to test

**Comment by tehsis** [comment]
Any news on this one?

File: lib/auth/auth.go [hunk 1]
```diff
@@ -673,11 +673,32 @@ func (a *AuthServer) ValidateOIDCAuthCallback(q url.Values) (*OIDCAuthResponse,
 		return response, nil
 	}
 
-	user, err := a.Identity.GetUserByOIDCIdentity(services.OIDCIdentity{
+	var user services.User
+
+	user, err = a.Identity.GetUserByOIDCIdentity(services.OIDCIdentity{
```
Comment (klizhentas) [inline lib/auth/auth.go]: I think this change is out of scope of this PR and should be submitted separately. So please revert this commit from this PR.

 Can you also write down a small issue/proposal explaining the desired change so we can discuss it separately.

**Review by klizhentas — CHANGES_REQUESTED** [review]

**Comment by tehsis** [comment]
@klizhentas fixed! I've created the issue to discuss this change.

**Comment by alex-kovoy** [comment]
Looks good to me too

**Review by alex-kovoy — APPROVED** [review]
Looks good

**Comment by klizhentas** [comment]
@tehsis please update with master and you are good to merge

**Review by klizhentas — APPROVED** [review]

**Comment by tehsis** [comment]
done!

**Comment by klizhentas** [comment]
ok, thanks for your contribution!

**Pull request merged**